src/vfvm_system.jl

```diff
@@ -103,10 +103,14 @@ mutable struct System{Tv, Tc, Ti, Tm, TSpecMat <: AbstractMatrix} <: AbstractSys
     generic_matrix::SparseMatrixCSC
 
     """
-    Sparse matrix colors for generic operator handling
+    Sparse matrix colors etc. for generic operator handling
     """
-    generic_matrix_colors::Vector
+    generic_matrix_prep::DifferentiationInterface.SparseJacobianPrep
 
+    """
+    Sparsity detector backend
+    """
+    generic_matrix_backend::DifferentiationInterface.AutoSparse
```

Comment:
Same remark about type instability, you could do
```suggestion
    generic_matrix_backend::B
```
and parametrize `B<:AutoSparse`

Reply:
see above, thanks